boxes/generators/can_storage.py

```diff
@@ -0,0 +1,232 @@
+        if self.top_edge != "e":
+            self.corner(90, self.thickness)
+            self.edge(0.5 * self.canDiameter)            
+            self.corner(-90, 0.25 * self.canDiameter)
+        else:
+            self.moveTo(-self.burn, self.canDiameter + self.thickness, -90)
+            self.corner(90, 0.25 * self.canDiameter)
+            self.edge(self.thickness)
+
+        self.edge(0.5 * self.canDiameter - self.thickness)
+        self.corner(-90, 0.25 * self.canDiameter)
+        self.edge(0.5 * self.canDiameter)
+        self.corner(90, self.thickness)
+        self.edge(x * self.thickness )
+        self.corner(90, self.thickness)
+        self.edge(0.5 * self.canDiameter)
+        self.corner(-90, 0.25 * self.canDiameter)
+        self.edge(0.5 * self.canDiameter - (1 + x) * self.thickness + self.top_chute_height + self.bottom_chute_height - self.barrier_height)
+        self.corner(-90, 0.25 * self.canDiameter)
+        self.edge(0.5 * self.canDiameter)
+        self.corner(90, self.thickness)
+        self.edge(self.barrier_height)
+        self.edge(self.thickness)
+
+class CanStorage(Boxes):
+    """Storage box for round containers"""
+
+    ui_group = "Misc"
+    
+    angle = 3.0 # slope angle of the chutes in degree
+    
+    def __init__(self):
+        Boxes.__init__(self)
+
+        self.addSettingsArgs(edges.FingerJointSettings, finger=2.0, space=2.0, surroundingspaces=0.0)
+        self.addSettingsArgs(fillHolesSettings)
+
+        self.argparser.add_argument(
+            "--top_edge", action="store",
+            type=ArgparseEdgeType("efhŠ"), choices=list("efhŠ"),
+            default="Š", help="edge type for top edge")
+        self.argparser.add_argument(
+            "--bottom_edge", action="store",
+            type=ArgparseEdgeType("eEš"), choices=list("eEš"),
+            default="š", help="edge type for bottom edge")
+
+        # Add non default cli params if needed (see argparse std lib)
+        self.argparser.add_argument(
+            "--canDiameter",  action="store", type=float, default=67,
+            help="outer diameter of the cans to be stored (in mm)")
+        self.argparser.add_argument(
+            "--canHight",  action="store", type=float, default=115,
+            help="hight of the cans to be stored (in mm)")
+        self.argparser.add_argument(
+            "--canNum",  action="store", type=int, default=12,
+            help="number of cans to be stored")
+    
+    def cb_top_chute(self, nr):
+        if nr == 0:
+            # fill with holes
+            border = [
+                (0, 0), 
+                (self.top_chute_depth, 0), 
+                (self.top_chute_depth, self.width), 
+                (0, self.width),                
+                ]
+            if self.mydebug:
+                self.showBorderPoly(border)
+            if self.fillHoles_fill_pattern != "no fill":
+                self.fillHoles(
+                    pattern="hbar", 
```

Comment:
The long slots make those chutes very fragile especially of the cans don't rest on the edges of the chute that are connected to the side walls. Beverages are heavy and you may run into all kind of trouble with the chute bending or even breaking.

Reply:
changed in PR #400